src/Enable.Extensions.Queuing.RabbitMQ/Internal/RabbitMQQueueClient.cs

```diff
@@ -24,13 +25,22 @@ internal class RabbitMQQueueClient : BaseQueueClient
         private bool _disposed;
 
         public RabbitMQQueueClient(
-            ConnectionFactory connectionFactory,
+            RabbitMQConnectionManager connectionManager,
             string queueName,
             QueueMode queueMode = QueueMode.Default)
         {
-            _connectionFactory = connectionFactory;
-            _connection = _connectionFactory.CreateConnection();
-            _channel = _connection.CreateModel();
+            _connectionFactory = connectionManager;
+            _connection = connectionManager.GetOrCreateConnection();
+
+            try
+            {
+                _channel = _connection.CreateModel();
+            }
+            catch (ChannelAllocationException)
+            {
+                _connection = connectionManager.CreateConnection();
+                _channel = _connection.CreateModel();
+            }
```

Comment:
There is quite a lot going on in this constructor. Constructors should be quick, and potentially this is creating up to two connections, and handling a thrown exception. Could we simplify this, perhaps by lazily creating the connection/channel when it's needed, rather than in the constructor (we might not want to do this if we're relying on some validation that occurs here), or by moving some of this complexity into the `RabbitMQConnectionManager` class?